components/src/jsMain/kotlin/dev/fritz2/components/alert.kt

```diff
@@ -0,0 +1,280 @@
+package dev.fritz2.components
+
+import dev.fritz2.binding.RootStore
+import dev.fritz2.binding.SimpleHandler
+import dev.fritz2.components.validation.ComponentValidationMessage
+import dev.fritz2.components.validation.Severity
+import dev.fritz2.dom.html.RenderContext
+import dev.fritz2.styling.StyleClass
+import dev.fritz2.styling.params.BasicParams
+import dev.fritz2.styling.params.styled
+import dev.fritz2.styling.theme.*
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOf
+
+/**
+ * A component to display an alert consisting of an icon, title and description.
+ * Different styles based on severities are supported, as well as a number of different layout options.
+ *
+ * Currently the following severities are available:
+ * - Info
+ * - Success
+ * - Warning
+ * - Error
+ * Specifying a severity will change the alert's color scheme based on the colors defined in the application theme.
+ * If no severity is specified, 'info' will be used by default.
+ *
+ * Additionally, a number of different layout options are available. These are:
+ * - 'subtle': A subtle style using different shades of the severity's base color defined in the application theme.
+ * - 'solid': A solid style using the severity's color from the application theme and a solid white color for the icon,
+ * text and decorations.
+ * - 'Top-Accent': A variation of the 'subtle' variant with a decoration element at the top.
+ * - 'Left-Accent': A variation of the 'subtle' variant with a decoration element on the left.
+ * If no variant is specified, 'solid' is used by default.
+ *
+ * Usage examples:
+ * ```
+ * alert {
+ *     title("Alert")
+ *     content("This is an alert.")
+ *     severity { info }
+ * }
+ *
+ * alert {
+ *     title("Alert")
+ *     content("This is an alert.")
+ *     severity { error }
+ *     variant { leftAccent }
+ * }
+ * ```
+ */
+@ComponentMarker
+class AlertComponent {
+
+    companion object {
+        private const val accentDecorationThickness = "4px"
+    }
```

Comment:
Does this component need size options, and with it a changing thickness value?

Reply:
At least there is no such option in Chakra UI, I believe. It would be very easy to add the option, though. What do you think?